mlir-sycl/test/Conversion/SYCLToLLVM/sycl-grid-ops-to-llvm.mlir

```diff
@@ -20,292 +20,278 @@ module attributes {gpu.container_module} {
     //CHECK-DAG:      llvm.mlir.global external constant @__builtin_var_GlobalInvocationId__() {addr_space = 0 : i32} : vector<3xi64>
     //CHECK-DAG:      llvm.mlir.global external constant @__builtin_var_NumWorkgroups__() {addr_space = 0 : i32} : vector<3xi64>
+    //CHECK-DAG:      llvm.mlir.global external constant @__builtin_var_GlobalSize__() {addr_space = 0 : i32} : vector<3xi64>
   gpu.module @kernels {
     // CHECK-LABEL:   llvm.func @test_num_work_items() -> !llvm.struct<"class.sycl::_V1::range.1", (struct<"class.sycl::_V1::detail::array.1", (array<1 x i64>)>)> {
-    // CHECK-DAG:         %[[VAL_0:.*]] = llvm.mlir.null : !llvm.ptr<struct<"class.sycl::_V1::range.1", (struct<"class.sycl::_V1::detail::array.1", (array<1 x i64>)>)>>
-    // CHECK-DAG:         %[[VAL_1:.*]] = llvm.mlir.constant(1 : index) : i64
-    // CHECK-NEXT:        %[[VAL_2:.*]] = llvm.getelementptr %[[VAL_0]][1] : (!llvm.ptr<struct<"class.sycl::_V1::range.1", (struct<"class.sycl::_V1::detail::array.1", (array<1 x i64>)>)>>) -> !llvm.ptr<struct<"class.sycl::_V1::range.1", (struct<"class.sycl::_V1::detail::array.1", (array<1 x i64>)>)>>
-    // CHECK-NEXT:        %[[VAL_3:.*]] = llvm.ptrtoint %[[VAL_2]] : !llvm.ptr<struct<"class.sycl::_V1::range.1", (struct<"class.sycl::_V1::detail::array.1", (array<1 x i64>)>)>> to i64
-    // CHECK-NEXT:        %[[VAL_4:.*]] = llvm.alloca %[[VAL_3]] x !llvm.struct<"class.sycl::_V1::range.1", (struct<"class.sycl::_V1::detail::array.1", (array<1 x i64>)>)> : (i64) -> !llvm.ptr<struct<"class.sycl::_V1::range.1", (struct<"class.sycl::_V1::detail::array.1", (array<1 x i64>)>)>>
-    // CHECK-DAG:         %[[VAL_5:.*]] = llvm.mlir.constant(0 : index) : i64
-    // CHECK-DAG:         %[[VAL_6:.*]] = llvm.mlir.constant(0 : i32) : i32
-    // CHECK-NEXT:        %[[VAL_7:.*]] = llvm.mlir.addressof @__builtin_var_NumWorkgroups__ : !llvm.ptr<vector<3xi64>>
-    // CHECK-NEXT:        %[[VAL_8:.*]] = llvm.load %[[VAL_7]] : !llvm.ptr<vector<3xi64>>
-    // CHECK-DAG:         %[[VAL_9:.*]] = llvm.mlir.constant(0 : i32) : i32
-    // CHECK-NEXT:        %[[VAL_10:.*]] = llvm.extractelement %[[VAL_8]]{{\[}}%[[VAL_9]] : i32] : vector<3xi64>
-    // CHECK-NEXT:        %[[VAL_11:.*]] = llvm.getelementptr inbounds %[[VAL_4]][0, 0, 0, 0] : (!llvm.ptr<struct<"class.sycl::_V1::range.1", (struct<"class.sycl::_V1::detail::array.1", (array<1 x i64>)>)>>) -> !llvm.ptr<i64>
-    // CHECK-NEXT:        llvm.store %[[VAL_10]], %[[VAL_11]] : !llvm.ptr<i64>
-    // CHECK-NEXT:        %[[VAL_12:.*]] = llvm.load %[[VAL_4]] : !llvm.ptr<struct<"class.sycl::_V1::range.1", (struct<"class.sycl::_V1::detail::array.1", (array<1 x i64>)>)>>
-    // CHECK-NEXT:        llvm.return %[[VAL_12]] : !llvm.struct<"class.sycl::_V1::range.1", (struct<"class.sycl::_V1::detail::array.1", (array<1 x i64>)>)>
+    // CHECK:             %[[VAL_0:.*]] = llvm.mlir.addressof @__builtin_var_GlobalSize__ : !llvm.ptr<vector<3xi64>>
```

Comment:
Why are we no longer using CHECK-NEXT and CHECK-DAG?

Reply:
Yes and no, it takes more time to use CHECK-NEXT than CHECK but CHECK-NEXT guarantees that no code will "fall" in between 2 CHECKS and so it can prevent more regressions. The drawback in using CHECK-NEXT is that some operations might be moved without changing the meaning of the program, but in those cases one should have used CHECK-DAG to start with. 

I do not have a strong preference but I'd say, in principle, using CHECK-NEXT is more regression proof.

Reply:
I wish the MLIR tool that generates the test would be as good as the LLVM tool.

Reply:
I'm using `CHECK-NEXT` now. No problem